tests/unit/test_collector.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python3
+"""Test collctor daemon."""
+import pytest
+
+
+class TestCollectorDaemon:
+    """Collector daemon test class."""
+
+    def test_fixture(self, collector_daemon):
+        """See if the helper fixture works to load charm configs."""
+        stats_collector_daemon = collector_daemon()
+        assert stats_collector_daemon is not None
+
+    def test_parse_config(self, collector_daemon):
+        """Test config parsing."""
+        statsd = collector_daemon()
+        assert (
+            statsd.config["juju"]["controller_endpoint"].get() == "192.168.1.100:17070"
+        )
+        assert (
+            statsd.config["juju"]["controller_cacert"].get()
+            == "-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----\n"
+        )
+        assert statsd.config["juju"]["username"].get() == "example_user"
+        assert statsd.config["juju"]["password"].get() == "example_password"
+        assert statsd.config["customer"]["name"].get() == "example_customer"
+        assert statsd.config["customer"]["cloud_name"].get() == "example_cloud"
+
+    @pytest.mark.asyncio
+    async def test_get_stats(self, collector_daemon):
+        """Test get_stats function and the execution of the collector."""
+        statsd = collector_daemon()
+        await statsd.get_stats()
```

Comment:
Should we test (assert) something here? It seems not test any function here.

And I think
* create_metrics_dict
* get_stats
These two functions are the most important functions in collector. 
But we do not verify the format of metrics here.

Reply:
Hi Eric. Thanks for the review. I've added an assert statement in #4 for this function (because that function is dedicated for the collector portion). Checking the correctness of label created by running `get_stats` would be enough to verify the code. IMO checking `create_metrics_dict` in unittests is not needed because the Gauge instance is entirely mocked.